### ISSUE TITLE
IME input does not work, removes extra characters

### ISSUE THREAD

**Issue by pizn** [ianstormtaylor/slate#854]
As follows:

![input-bug](https://cloud.githubusercontent.com/assets/986480/26564088/1c7199aa-450f-11e7-88b9-65294af04a63.gif)

I use Chrome on OSX.

**Comment by yuanyong** [comment]
I've got the same issue too.

**Comment by yuanyong** [comment]
The issue is just happing when the input source is simple Chinese.

**Comment by pizn** [comment]
@ianstormtaylor I try to solve the IME question I have got in my RP #880 .

It's a difficult problem for IME, I don't want it always re-render when user input, but it have to re-render when react state is update and the DOM is old. Be exact, the DOM is empty(null) and should forceUpdate() to re-render.

Composition Event and the unreconcilable state is the problem，maybe I only solved the surface of the problem.
